test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -153,5 +176,544 @@ public async Task DiscoverAsync_IsIdempotent()
+        public void IsValidPluginFile_Unix_ExecutableFile_ReturnsTrue()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath).Dispose();
+
+#if NET8_0_OR_GREATER
+            // Set execute permissions
+            File.SetUnixFileMode(pluginFilePath, UnixFileMode.UserExecute | UnixFileMode.UserRead);
+#else
+            // Use chmod to set execute permissions
+            var process = new Process();
+            process.StartInfo.FileName = "/bin/bash";
+            process.StartInfo.Arguments = $"-c \"chmod +x {pluginFilePath}\"";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.Start();
+            process.WaitForExit();
+#endif
+
+            var fileInfo = new FileInfo(pluginFilePath);
+
+            // Act
+            bool result = PluginDiscoverer.IsValidPluginFile(fileInfo);
+
+            // Assert
+            Assert.True(result);
+        }
+
+#if !NET8_0_OR_GREATER
+        [PlatformFact(Platform.Linux)]
+        public void IsExecutable_FileIsExecutable_ReturnsTrue()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath);
+
+            // Set execute permissions
+            var process = new Process();
+            process.StartInfo.FileName = "/bin/bash";
+            process.StartInfo.Arguments = $"-c \"chmod +x {pluginFilePath}\"";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.Start();
+            process.WaitForExit();
+
+            var fileInfo = new FileInfo(pluginFilePath);
+
+            // Act
+            bool result = PluginDiscoverer.IsExecutable(fileInfo);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [PlatformFact(Platform.Linux)]
+        public void IsExecutable_FileIsNotExecutable_ReturnsFalse()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath);
+
+            // Remove execute permissions
+            var process = new Process();
+            process.StartInfo.FileName = "/bin/bash";
+            process.StartInfo.Arguments = $"-c \"chmod -x {pluginFilePath}\"";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.Start();
+            process.WaitForExit();
```

Comment:
Given that WaitForExit waits indefinitely for the process to exit, you might want to consider addressing [this comment](https://github.com/NuGet/NuGet.Client/pull/5990/files#r1811192719) by changing it to the following:

```cs
Assert.True(process.WaitForExit(1000), userMessage: "some string that helps to troubleshoot the issue");
```

With this approach, the test fails gracefully if the process doesn't complete on time for any unknown reason.